language-plutus-core/create-cost-model/CreateModel.hs

```diff
@@ -0,0 +1,143 @@
+      paramDivideInteger <- divideInteger benchData
+      paramQuotientInteger <- quotientInteger benchData
+      paramRemainderInteger <- remainderInteger benchData
+      paramModInteger <- modInteger benchData
+      paramLessThanInteger <- lessThanInteger benchData
+      paramLessThanEqInteger <- lessThanEqInteger benchData
+      paramGreaterThanInteger <- greaterThanInteger benchData
+      paramGreaterThanEqInteger <- greaterThanEqInteger benchData
+      paramEqInteger <- eqInteger benchData
+      paramConcatenate <- concatenate benchData
+      paramTakeByteString <- takeByteString benchData
+      paramDropByteString <- dropByteString benchData
+      paramSHA2 <- sHA2 benchData
+      paramSHA3 <- sHA3 benchData
+      paramVerifySignature <- verifySignature benchData
+      paramEqByteString <- eqByteString benchData
+      paramLtByteString <- ltByteString benchData
+      paramGtByteString <- gtByteString benchData
+      paramIfThenElse <- ifThenElse benchData
+
+      pure $ CostModel {..}
+  BSL.writeFile "language-plutus-core/src/costModel.json" $ encodePretty' (defConfig { confCompare = \_ _-> EQ }) model
+
+filterDF :: MonadR m => Text -> (SomeSEXP (Region m)) -> m (SomeSEXP (Region m))
+filterDF by df =
+  [r| df_hs %>% filter(BuiltinName %in% c(by_hs))|]
+
+data LinearModelRaw = LinearModelRaw
+  { linearModelRawTerm      :: String
+  , linearModelRawEstimate  :: Double
+  , linearModelRawStdError  :: Double
+  , linearModelRawStatistic :: Double
+  , linearModelRawPValue    :: Double
+  } deriving (Show, Eq)
+
+instance FromJSON LinearModelRaw where
+    parseJSON = withObject "LinearModelRaw" $ \v -> LinearModelRaw
+        <$> v .: "term"
+        <*> v .: "estimate"
+        <*> v .: "std.error"
+        <*> v .: "statistic"
+        <*> v .: "p.value"
+
+findInRaw :: String -> Vector LinearModelRaw -> Either String LinearModelRaw
+findInRaw s v = maybeToEither ("Couldn't find the term " <> s) $
+  Data.Vector.find (\e -> linearModelRawTerm e == s) v
+
+linearAdditiveModel :: MonadR m => (SomeSEXP (Region m)) -> m (ModelAddedSizes)
+linearAdditiveModel df = do
+  j <- [r| jsonlite::toJSON(tidy(lm(Mean ~ I(x_mem + y_mem), df_hs))) |]
+  let m = do
+        model <- eitherDecodeStrict $ T.encodeUtf8 $ (fromSomeSEXP j :: Text)
+        intercept <- linearModelRawEstimate <$> findInRaw "(Intercept)" model
+        slope <- linearModelRawEstimate <$> findInRaw "I(x_mem + y_mem)" model
+        pure $ ModelAddedSizes intercept slope
+  case m of
+    Left err -> throwM (TypeError err)
+    Right x  -> pure x
+
+
+addInteger :: MonadR m => (SomeSEXP (Region m)) -> m (CostingFun ModelTwoArguments)
+addInteger df = do
+  addInt <- filterDF "AddInteger" df
+  cpuModel <- linearAdditiveModel addInt
+  pure $ CostingFun (ModelTwoArgumentsAddedSizes cpuModel) def
+
+subtractInteger :: MonadR m => (SomeSEXP (Region m)) -> m (CostingFun ModelTwoArguments)
+subtractInteger _ = pure def
```

Comment:
Not sure what's going on here.  Are all of these waiting to be filled in with something similar to the addInteger case?

Reply:
Exactly. I'll add in one or two more to make it clearer.